### PR TITLE
Preventing retentionTime and retentionCount to be set to 0 for dataset

### PR DESCRIPTION
**Pull request by SophieGuo410**

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java [hunk 1]
```diff
@@ -595,6 +595,10 @@ public static long getTtlFromRequestHeader(Map<String, Object> args) throws Rest
         throw new RestServiceException(Headers.TTL + "[" + ttlFromHeader + "] is not valid (has to be >= -1)",
             RestServiceErrorCode.InvalidArgs);
       }
+      if (ttlFromHeader == 0) {
+        throw new RestServiceException(Headers.TTL + "[" + ttlFromHeader + "] should not be 0",
+            RestServiceErrorCode.InvalidArgs);
+      }
```
Comment (justinlin-linkedin) [inline ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java:601]: we probably don't need this one here.

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java (hunk 1 above)
Comment (SophieGuo410) [reply ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java]: Remove this part and updated to add test and filter out null(default) after serialization as we discussed.

File: ambry-api/src/main/java/com/github/ambry/account/Dataset.java [hunk 2]
```diff
@@ -228,3 +232,30 @@ public String toString() {
...
+
+    // Use reflection to get all fields of the Dataset class
+    Field[] fields = Dataset.class.getDeclaredFields();
+    for (Field field : fields) {
+      if (field.isAnnotationPresent(JsonProperty.class)) {
+        try {
+          field.setAccessible(true);
+          Object value = field.get(dataset);
+          if (value != null) {
+            gen.writeObjectField(field.getName(), value);
```
Comment (justinlin-linkedin) [inline ambry-api/src/main/java/com/github/ambry/account/Dataset.java:252]: we should probably use annotation value here. something like
```
gen.writeObjectField(field.getAnnotation(JsonProperty.class).value(), value);
```

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-api/src/main/java/com/github/ambry/account/Dataset.java (hunk 2 above)
Comment (SophieGuo410) [reply ambry-api/src/main/java/com/github/ambry/account/Dataset.java]: Sure, updated it.
